TROUBLESHOOTING.md

```diff
@@ -0,0 +1,192 @@
+# Troubleshooting Common Issues
+
+## Using this Document
+
+Search for the error message you're getting and see if there's a match, or simply skim the table of contents below for topics that seem relevant to the issue you're having. Each issue section has steps to work around or fix the particular issue, and have examples of common error messages.
+
+If you do not find the issue below, try searching the issue tracker itself for potential duplicates before opening a new issue.
+
+If you're reading this document because you noticed an issue with npm's web site, you may want to let the [web team](https://github.com/npm/www/issues) know.
+
+### Updating this Document
+
+If you think something should be added here, make a PR that includes the following:
+
+0. a summary
+0. one or more example errors
+0. steps to debug and fix
+0. links to at least one related issue from the tracker
+
+For more details of the content and formatting of these entries, refer to examples below.
+
+## Table of Contents
+
+* [Upgrading npm](#upgrading-npm)
+* [Proxies and Networking](#proxy-and-networking-issues)
+* [Cannot find module](#cannot-find-module)
+* [Shasum Check Fails](#shasum-check-fails)
+* [No Git](#no-git)
+
```

Comment:
A post-merge PR on this would be nice to iterate on if anyone wants to volunteer.